components/Wallet/WalletForm.tsx

```diff
@@ -1,54 +1,170 @@
 import React, { ReactElement, useState, useEffect } from 'react'
 import { useForm } from 'react-hook-form'
-import { paybuttonPOSTParameters } from 'utils/validators'
+import { WalletPOSTParameters } from 'utils/validators'
+import { PaybuttonWithAddresses } from 'services/paybuttonService'
+import { XEC_NETWORK_ID, BCH_NETWORK_ID } from 'constants/index'
 import Image from 'next/image'
-import style from '../Paybutton/paybutton.module.css'
+import style from '../Wallet/wallet.module.css'
+import style_pb from 'components/Paybutton/paybutton.module.css'
 import Plus from 'assets/plus.png'
+import axios from 'axios'
+import { appInfo } from 'config/appInfo'
 
 interface IProps {
-  onSubmit: Function
-  paybuttons: []
-  error: String
-  editname: boolean
+  userPaybuttons: PaybuttonWithAddresses[]
+  refreshWalletList: Function
+  userId: string
 }
 
-export default function WalletForm ({ onSubmit, paybuttons, error, editname }: IProps): ReactElement {
-  const { register, handleSubmit, reset } = useForm<paybuttonPOSTParameters>()
+export default function WalletForm ({ userPaybuttons, refreshWalletList, userId }: IProps): ReactElement {
+  const { register, handleSubmit, reset } = useForm<WalletPOSTParameters>()
   const [modal, setModal] = useState(false)
+  const [isXECDefaultDisabled, setIsXECDefaultDisabled] = useState(true)
+  const [isBCHDefaultDisabled, setIsBCHDefaultDisabled] = useState(true)
+  const [error, setError] = useState('')
+  const [selectedPaybuttonIdList, setSelectedPaybuttonIdList] = useState([] as string[])
+
+  async function onSubmit (params: WalletPOSTParameters): Promise<void> {
+    params.userId = userId
+    try {
+      void await axios.post(`${appInfo.websiteDomain}/api/wallet`, params)
+      refreshWalletList()
+      setError('')
+    } catch (err: any) {
+      setError(err.response.data.message)
+    }
+  }
+
+  function handleSelectedPaybuttonsChange(checked: boolean, paybuttonId: string): void {
+    if (selectedPaybuttonIdList.includes(paybuttonId) && checked === false) {
+      setSelectedPaybuttonIdList(
+        selectedPaybuttonIdList.filter(id => id !== paybuttonId)
+      )
+    }
+    if (!selectedPaybuttonIdList.includes(paybuttonId) && checked === true) {
+      setSelectedPaybuttonIdList(
+        [...selectedPaybuttonIdList, paybuttonId]
+      )
+    }
+  }
+
+  function hasAddressForNetworkId(networkId: number): boolean {
+    let ret = false
+    if (selectedPaybuttonIdList === undefined) return false
+    for (const selectedPaybuttonId of selectedPaybuttonIdList) {
+      let paybutton = userPaybuttons.find((pb) => pb.id === Number(selectedPaybuttonId))
+      if (paybutton === undefined) {
```

Comment:
Can we avoid using `continue`? We could add this `if` condition to the next line:
```js
if (paybutton !== undefined && paybutton.addresses.some... )
```

Reply:
Addressed.